src/Core/Resolvers/SqlMutationEngine.cs

```diff
@@ -218,6 +218,44 @@ public class SqlMutationEngine : IMutationEngine
             return result;
         }
 
+        /// <summary>
+        /// Helper method to determine:
+        /// 1. Whether a mutation is a mutate one or mutate many operation (eg. createBook/createBooks)
+        /// 2. Name of the top-level entity backing the mutation.
+        /// </summary>
+        /// <param name="context">GraphQL request context.</param>
+        /// <returns>a tuple of the above mentioned metadata.</returns>
+        private static Tuple<bool, string> GetMutationCategoryAndEntityName(IMiddlewareContext context)
+        {
+            IOutputType outputType = context.Selection.Field.Type;
+            string entityName = string.Empty;
+            ObjectType _underlyingFieldType = GraphQLUtils.UnderlyingGraphQLEntityType(outputType);
+            bool isPointMutation;
+            if (GraphQLUtils.TryExtractGraphQLFieldModelName(_underlyingFieldType.Directives, out string? modelName))
+            {
+                isPointMutation = true;
+                entityName = modelName;
+            }
+            else
+            {
+                // Absence of model directive on the mutation indicates that we are dealing with a `mutate many`
+                // mutation like createBooks.
```

Comment:
Model directive is not added on `*Connection` object types (which is the input type for multiple create operation type). Hence, the conclusion.

It would be helpful to add more details (wherever applicable) on why the failure to meet a certain condition leads to a certain conclusion.

Reply:
Makes sense. I'll add the comment.